src/mlpack/methods/ann/layer/perametric_relu_impl.hpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file perametric_relu_impl.hpp
+ * @author Prasanna Patil
+ *
+ * Definition of PerametricReLU layer first introduced in the,
+ * Kaiming He, Xiangyu Zhang, Shaoqing, Ren Jian Sun,
+ * "Delving Deep into Rectifiers:
+ * Surpassing Human-Level Performance on ImageNet Classification", 2014
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_PERAMETRICRELU_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_PERAMETRICRELU_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "perametric_relu.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+PerametricReLU<InputDataType, OutputDataType>::PerametricReLU(
+    const double user_alpha) : user_alpha(user_alpha)
+{
+  alpha.set_size(1,1);
```

Comment:
Alpha is a trainable. So it is passed when Parameters() method is called and WeightSetVisitor, WeightSizeVisitor etc.. Parameters() to be an arma::mat. So how to pass double rather than arma::mat?

We can create some other matrix (i.e. weigh) then create a double pointer (or reference) which references to first element of weight but I don't know if that would work or not. What do you propose?